packages/crayons-core/src/utils/select-utils.ts

```diff
@@ -0,0 +1,221 @@
+        meta: {
+          shouldValidate: false, // for handling validation with form during reset. watcher in list-options is firing.
+          selectedOptions: this.selectedOptionsState,
+        },
+      });
+    }
+  }
+}
+
+export function expandHandler(expanded) {
+  const icon = this.host.shadowRoot
+    ?.querySelector('.select-container')
+    ?.querySelector('fw-button')
+    ?.shadowRoot?.querySelector('fw-icon');
+  icon && (icon.name = expanded ? 'chevron-up' : 'chevron-down');
+}
+
+export function initializeComponent() {
+  //TODO: The below is a rough draft and needs to be optimized for better performance.
+  const selectOptions = Array.from(
+    this.host.querySelectorAll('fw-select-option')
+  );
+
+  // Set value if the selectedOptions is provided
+  if (this.selectedOptions.length > 0) {
+    this.selectedOptionsState = this.selectedOptions;
+    this.value = this.multiple
+      ? this.selectedOptions.map((option) => option.value)
+      : this.selectedOptions[0].value;
+  }
+
+  if (this.multiple) {
+    if (this.multiple && typeof this.value === 'string') {
+      throw Error('value must be a array of string when multiple is true');
+    }
+    this.value = this.value?.length > 0 ? this.value : [];
+  } else {
+    this.value = this.value ? this.value : '';
+  }
+  const options = selectOptions.map((option: any) => {
+    return {
+      html: option.html,
+      text: option.html ? option.optionText : option.textContent,
+      value: option.value,
+      selected: this.isValueEqual(this.value, option) || option.selected,
+      disabled: option.disabled,
+      htmlContent: option.html ? option.innerHTML : '',
+    };
+  });
+  this.dataSource = options.length === 0 ? this.options : options;
+  // Set selectedOptions if the value is provided
+  if (!this.multiple && this.value && this.selectedOptions.length === 0) {
+    this.selectedOptionsState = this.dataSource.filter(
+      (option) => this.value === option.value
+    );
+  } else if (
+    this.multiple &&
+    this.value.length !== this.selectedOptions.length
+  ) {
+    this.selectedOptionsState = this.dataSource.filter((option) =>
+      this.isValueEqual(this.value, option)
+    );
+  }
+  if (this.dataSource?.length > 0) {
+    // Check whether the selected data in the this.dataSource  matches the value
+    const selectedDataSource = this.dataSource.filter(
+      (option) => option.selected
+    );
+    const selectedDataSourceValues = selectedDataSource.map(
+      (option) => option.value
+    );
+    const selected = this.multiple
+      ? selectedDataSourceValues
+      : selectedDataSourceValues[0];
+    if (
+      selectedDataSourceValues.length > 0 &&
+      JSON.stringify(this.value) !== JSON.stringify(selected)
+    ) {
+      this.value = selected;
+      this.selectedOptionsState = selectedDataSource;
+    }
+  }
+  this.host.addEventListener('focus', this.setFocus);
+
+  //Get id
+  this.hostId = this.host.id || '';
+}
```

Comment:
This whole Init function looks like a big bundle of code. Am not able to clearly understand whats happening here. Please discuss this in a call.